### PR TITLE
fix(route): update The Inithium authentication method

### LINKED ISSUE #13388: The Initium 端传媒 router fails to fetch full text article behind paywall
### Routes

```routes
/theinitium/channel/:type?/:language?
```

### Full routes

```fullroutes
/theinitium/channel/latest/zh-hans
```

### Related documentation

https://docs.rsshub.app/routes/new-media#duan-chuan-mei-zhuan-ti-lan-mu

### What is expected?

With default ENV variable or personal variable like `INITIUM_BEARER_TOKEN`, the router should be able to fetch full-text article from the source.

### What is actually happening?

Some articles failed to be fetched in full-text mode, showing for example :

CleanShot 2023-09-25 at 14 28 18

### Deployment information

Self-hosted

### Deployment information (for self-hosted)

_No response_

### Additional info

```shell
No relative logs found regarding this issue.
```

### This is not a duplicated issue

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) to ensure this bug has not already been reported

### PR DESCRIPTION
**Pull request by mintyfrankie**
## Involved Issue / 该 PR 相关 Issue

Close #13388 

## Example for the Proposed Route(s) / 路由地址示例

```routes
/theinitium/channel/latest/zh-hans
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

- Remove authentication method by `INITIUM_IAP_RECEIPT`, as the iOS app no longer use header `X-IAP-Receipt` in its requests
- Fix authentication method by username and password login
- Add maintainer

### CODE REVIEW

**Comment by mintyfrankie** [comment]
PR related to #13388 

Questions:

1. The route `/theinitium/follow/articles/:language?` seems not to be working anymore; should I create another PR or it's fine that I add the fix in this one?
2. The explanation about The Inithium on the **Deployment Page** will have to be updated since this PR rules out the need of `INITIUM_IAP_RECEIPT`. How can we proceed to this?

Thanks!

File: lib/routes/theinitium/utils.ts [hunk 1]
```diff
@@ -39,7 +39,6 @@ export const processFeed = async (model: string, ctx: Context) => {
     const key = {
         email: config.initium.username,
         password: config.initium.password,
-        iapReceipt: config.initium.iap_receipt,
```
Comment (TonyRL) [inline lib/routes/theinitium/utils.ts:42]: https://github.com/DIYgod/RSSHub/blob/0f01fe60cbc6fc57f8fa01e258267954b2110d1f/lib/routes/theinitium/utils.ts#L144
is still using key.iapReceipt

Please remove corresponding entries in `lib/config.ts` as well.

File: lib/routes/theinitium/utils.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import { Context } from 'hono';
+import { config } from '@/config';
+import InvalidParameterError from '@/errors/types/invalid-parameter';
```
[the remark this answers is not in the document]
Comment (mintyfrankie) [reply lib/routes/theinitium/utils.ts:2]: This was automatically formatted by prettier based on the `.prettierrc`. Should I still revert to the previous order?

File: lib/routes/theinitium/utils.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 import cache from '@/utils/cache';
 import got from '@/utils/got';
-import { config } from '@/config';
-import { load } from 'cheerio';
 import { parseDate } from '@/utils/parse-date';
-import InvalidParameterError from '@/errors/types/invalid-parameter';
+import { load } from 'cheerio';
+import { Context } from 'hono';
```
[the remark this answers is not in the document]
Comment (mintyfrankie) [reply lib/routes/theinitium/utils.ts:7]: Idem

File: lib/routes/theinitium/utils.ts (hunk 1 above)
Comment (mintyfrankie) [reply lib/routes/theinitium/utils.ts:42]: Thanks, entry removed.

**Comment by mintyfrankie** [comment]
> Questions:
> 
> 1. The route `/theinitium/follow/articles/:language?` seems not to be working anymore; should I create another PR or it's fine that I add the fix in this one?
> 2. The explanation about The Inithium on the **Deployment Page** will have to be updated since this PR rules out the need of `INITIUM_IAP_RECEIPT`. How can we proceed to this?
> 
> Thanks!

Could you please also take a look at these two questions as well?

File: lib/config.ts [hunk 4]
```diff
@@ -154,7 +154,6 @@ export type Config = {
         username?: string;
         password?: string;
         bearertoken?: string;
-        iap_receipt?: string;
```
Comment (TonyRL) [inline lib/config.ts:157]: https://github.com/DIYgod/RSSHub/blob/2a2a3066bfbc9d59e0632bff5575d960bd13a7b9/lib/config.ts#L508-L513
is still not being removed

Comment (mintyfrankie) [reply lib/config.ts:157]: Nice catch! My bad

**Comment by mintyfrankie** [comment]
Well just checked and the API used by the other route seems to be deprecated. Can you please review and merge what we have for the time being please?

**Comment by mintyfrankie** [comment]
Sorry just found a bug. Will work on it.

**Comment by mintyfrankie** [comment]
Should be good now. Thanks!

**Pull request merged**